platform/security/sts/security-sts-anonymousclaimshandler/src/main/java/org/codice/ddf/security/claims/anonymous/AnonymousClaimsHandler.java

```diff
@@ -134,6 +138,22 @@ public ProcessedClaimCollection retrieveClaimValues(ClaimCollection claims,
             }
         }
 
+        if (principal != null) {
+            String ipAddress = AnonymousPrincipal.parseAddressFromName(principal.getName());
```

Comment:
This is going to break on Kerberos principals. They separate the name and realm with \@.